tensorrt_llm/llmapi/llm_args.py

```diff
@@ -955,11 +950,43 @@ class BaseLlmArgs(BaseModel):
         default=None,
         description="The optional MPI session to use for this LLM instance.",
         json_schema_extra={"type": "Optional[MpiSession]"},
-        exclude=True,  # exclude from serialization
+        exclude=True,
         alias="_mpi_session")
 
+    backend: Optional[str] = Field(
+        default=None,
+        description="The backend to use for this LLM instance.",
+        exclude_json_schema=True,  # hide from API references
+    )
+
+    _parallel_config: Optional[object] = PrivateAttr(default=None)
+    _model_format: Optional[_ModelFormatKind] = PrivateAttr(default=None)
+    _speculative_model: Optional[str] = PrivateAttr(default=None)
+    _speculative_model_format: Optional[_ModelFormatKind] = PrivateAttr(
+        default=None)
+
+    @property
+    def parallel_config(self) -> _ParallelConfig:
+        return self._parallel_config
+
+    @property
+    def model_format(self) -> _ModelFormatKind:
+        return self._model_format
+
+    @property
+    def speculative_model(self) -> Optional[_ModelFormatKind]:
+        return self._speculative_model
+
+    @property
+    def speculative_model_format(self) -> _ModelFormatKind:
+        return self._speculative_model_format
+
     @print_traceback_on_error
     def model_post_init(self, __context: Any):
+        self._ensure_lora_config_consistency()
+
+        self.max_input_len = self.max_input_len or 1024
+        self.quant_config = self.quant_config or QuantConfig()
```

Comment:
Same here. why not use a field validator? 

Generally speaking, I suggest we keep the `model_post_init` function at a minimum to avoid obfuscation. Field validators are supposed to provide a more scalable, modular approach to post-processing field values in pydantic models

Reply:
Good suggestion, updated.